zapcore/field_test.go

```diff
@@ -320,6 +320,16 @@ func TestEquals(t *testing.T) {
 			b:    zap.Dict("k", zap.String("a", "d")),
 			want: false,
 		},
+		{
+			a:    zap.Object("k", zap.DictObject(zap.String("a", "b"))),
+			b:    zap.Object("k", zap.DictObject(zap.String("a", "b"))),
+			want: true,
+		},
+		{
+			a:    zap.Object("k", zap.DictObject(zap.String("a", "b"))),
```

Comment:
The example test is good, and this equality check makes sense.
Would you also mind adding a test similar to [TestDict](https://github.com/uber-go/zap/blob/3f27eb9c787ea9b2ddfc1f707ef291c9070655fe/field_test.go#L294) that asserts the shape of the resulting JSON-ey object.

Reply:
sounds good, I put one here: https://github.com/uber-go/zap/pull/1460/files#diff-218ab55776e43f87dc44a4b65eebeff94670cebeb5dcf80ca3143b5577b50e7dR318